csharp/ql/src/semmle/code/csharp/controlflow/ControlFlowGraph.qll

```diff
@@ -842,80 +864,33 @@ module ControlFlow {
           )
         or
         // Post-order: element itself
-        cfe instanceof StandardExpr and
+        cfe instanceof PostOrderElement and
         result = cfe and
         c = getValidSelfCompletion(result)
         or
         // Pre/post order: a child exits abnormally
         result = cfe.(StandardElement).getChildElement(_) and
         c = TRec(TLastRecAbnormalCompletion())
         or
-        cfe =
-          any(LogicalNotExpr lne |
-            // Operand exits with a Boolean completion
-            result = lne.getOperand() and
-            c = TRec(TLastRecBooleanNegationCompletion())
-            or
-            // Operand exits with a non-Boolean completion
-            result = lne.getOperand() and
-            c = TRec(TLastRecNonBooleanCompletion())
-          )
+        // Operand exits abnormally
+        result = cfe.(LogicalNotExpr).getOperand() and
+        c = TRec(TLastRecAbnormalCompletion())
```

Comment:
Why do we need to list this many cases for the abnormal completion? Isn't it the case that any child (of tyoe `Expr`) can exit abnormally?

Reply:
This should be better in an upcoming refactoring.